zzz_generated/app-common/app-common/zap-generated/cluster-objects.h

```diff
@@ -0,0 +1,1507 @@
+enum class NavigateTargetStatus : uint8_t
+{
+    Success         = 0x00,
+    AppNotAvailable = 0x01,
+    SystemBusy      = 0x02,
+};
+
+namespace NavigateTargetTargetInfo {
+enum FieldId
+{
+    kIdentifierFieldId = 0;
+
+    kNameFieldId = 1;
+
+};
+struct Type
+{
+    uint8_t identifier;
+    chip::ByteSpan name;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+} // namespace for NavigateTargetTargetInfo
+} // namespace for Target Navigator
+namespace TemperatureMeasurement {
+constexpr ClusterId kClusterId = 0x402;
+
+} // namespace for Temperature Measurement
+namespace TestCluster {
+constexpr ClusterId kClusterId = 0x50F;
+// Enum for SimpleEnum
+enum class SimpleEnum : uint8_t
+{
+    Unspecified = 0x00,
+    ValueA      = 0x01,
+    ValueB      = 0x02,
+    ValueC      = 0x03,
+};
+
+namespace SimpleStruct {
+enum FieldId
+{
+    kAFieldId = 0;
+
+    kBFieldId = 1;
+
+    kCFieldId = 2;
+
+    kDFieldId = 3;
+
+    kEFieldId = 4;
+
+};
+struct Type
+{
+    uint8_t a;
+    bool b;
+    SimpleEnum c;
+    chip::ByteSpan d;
+    Span<const char> e;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+} // namespace for SimpleStruct
+namespace NestedStruct {
+enum FieldId
+{
+    kAFieldId = 0;
+
+    kBFieldId = 1;
+
+    kCFieldId = 2;
+
+};
+struct Type
+{
+    uint8_t a;
+    bool b;
+    SimpleStruct::Type c;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+} // namespace for NestedStruct
+namespace NestedStructList {
+enum FieldId
+{
+    kAFieldId = 0;
+
+    kBFieldId = 1;
+
+    kCFieldId = 2;
+
+    kDFieldId = 3;
+
+    kEFieldId = 4;
+
+    kFFieldId = 5;
+
+    kGFieldId = 6;
+
+};
+struct Type
+{
+    uint8_t a;
+    bool b;
+    SimpleStruct::Type c;
+    DataModel::List<SimpleStruct::Type> d;
+    DataModel::List<uint32_t> e;
+    DataModel::List<chip::ByteSpan> f;
+    DataModel::List<uint8_t> g;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+struct DecodableType
+{
+    uint8_t a;
+    bool b;
+    SimpleStruct::Type c;
+    DataModel::DecodableList<SimpleStruct::Type> d;
+    DataModel::DecodableList<uint32_t> e;
+    DataModel::DecodableList<chip::ByteSpan> f;
+    DataModel::DecodableList<uint8_t> g;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+} // namespace for NestedStructList
+namespace DoubleNestedStructList {
+enum FieldId
+{
+    kAFieldId = 0;
+
+};
+struct Type
+{
+    DataModel::List<NestedStructList::Type> a;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+struct DecodableType
+{
+    DataModel::DecodableList<NestedStructList::Type> a;
```

Comment:
@mrjerryjohns  this line stucks....NestedStructList::Type ==> NestedStructList::DecodableType